geometry.py

```
@@ -0,0 +1,180 @@
+        (bl, tl)  # Left
+    ]
+
+def find_bezier_rect_intersection(p0: Point, p1: Point, p2: Point, p3: Point, 
+                                  rect_x: float, rect_y: float, rect_w: float, rect_h: float,
+                                  num_samples: int = 20) -> Optional[Point]:
+    """
+    Find the intersection of a Bezier curve and a rectangle.
+    Approximates the Bezier curve as line segments.
+    """
+    if num_samples < 1:
+        num_samples = 1
```

Ruff (0.15.6)

[warning] Replace `if` statement with `num_samples = max(num_samples, 1)`

Replace with `num_samples = max(num_samples, 1)`

(PLR1730)